pkg/reconciler/common/releases.go

```diff
@@ -63,6 +65,52 @@ func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
 	return TargetManifest(instance)
 }
 
+// IsUpDowngradeEligible returns the bool indicate whether the installed manifest is able to upgrade or downgrade to
+// the target manifest.
+func IsUpDowngradeEligible(instance v1alpha1.KComponent) bool {
+	current := instance.GetStatus().GetVersion()
+	// If there is no manifest installed, return true, because the target manifest is able to install.
+	if current == "" {
+		return true
+	}
+	current = sanitizeSemver(current)
+	target := sanitizeSemver(TargetVersion(instance))
+
+	currentMajor := semver.Major(current)
+	targetMajor := semver.Major(target)
+	if currentMajor != targetMajor {
+		// All the official releases of Knative are under the same Major version number. If target and current versions
+		// are different in terms of major version, upgrade or downgrade is not supported.
+		// TODO We need to deal with the the case of bumping major version later.
+		return false
+	}
+
+	var err error
```

Comment:
nit: I don't think you need to declare `err` here given the `:=` on the next line.